Userland/Libraries/LibCore/Platform/MachMessageTypes.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2024, Andrew Kaster <akaster@serenityos.org>
+ * Copyright (c) 2024, Aliaksandr Kalenik <kalenik.aliaksandr@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Platform.h>
+
+#if !defined(AK_OS_MACH)
+#    error "This file is only available on Mach platforms"
+#endif
+
+#include <mach/mach.h>
+
+namespace Core::Platform {
+
+struct MessageBodyWithSelfTaskPort {
+    mach_msg_body_t body;
+    mach_msg_port_descriptor_t port_descriptor;
+    mach_msg_audit_trailer_t trailer;
+};
+
+struct MessageWithSelfTaskPort {
+    mach_msg_header_t header;
+    mach_msg_body_t body;
+    mach_msg_port_descriptor_t port_descriptor;
```

Comment:
This is all untyped MIG-specific, unfortunately, so it will work on Darwin, but not on GNN Mach :cry: